File: docs-examples/example-java/src/test/java/io/micronaut/jms/docs/quickstart/QuickstartSpec.java
package io.micronaut.jms.docs.quickstart;

import io.micronaut.jms.docs.AbstractJmsSpec;
import org.junit.jupiter.api.Test;

import static java.util.concurrent.TimeUnit.SECONDS;
import static org.awaitility.Awaitility.await;

class QuickstartSpec extends AbstractJmsSpec {

    @Test
    void testTextProducerAndConsumer() {
// tag::producer[]
TextProducer textProducer = applicationContext.getBean(TextProducer.class);
textProducer.send("quickstart");
// end::producer[]

        TextConsumer textConsumer = applicationContext.getBean(TextConsumer.class);
        await().atMost(3, SECONDS).until(() ->
            textConsumer.messages.size() == 1 &&
                textConsumer.messages.get(0).equals("quickstart")
        );
    }
}

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/jms/docs/quickstart/QuickstartSpec.kt
package io.micronaut.jms.docs.quickstart

import io.micronaut.jms.docs.AbstractJmsSpec
import org.junit.jupiter.api.Test

import org.awaitility.Awaitility.await
import java.util.concurrent.TimeUnit

class QuickstartSpec : AbstractJmsSpec() {

    @Test
    fun testTextProducerAndConsumer() {
// tag::producer[]
val textProducer = applicationContext.getBean(TextProducer::class.java)
textProducer.send("quickstart")
// end::producer[]

        val textConsumer = applicationContext.getBean(TextConsumer::class.java)
        await().atMost(3, TimeUnit.SECONDS).until {
            textConsumer.messages.size == 1 &&
                textConsumer.messages[0] == "quickstart"
        }
    }
}
